content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,171 @@
+
+### Effects
+
+Effects and other listeners (such as memos) can be thought of as automated responders.
+They are closely tied to signals and are set into motion whenever there's a change in the signals they observe. 
+Their primary role is to ensure that any modifications in signals are appropriately reflected throughout the system, maintaining its reactivity.
+
+The nature of effects revolves around two functionalities:
+
+- **Observation**: At the heart of every effect is its ability to continuously monitor or "observe" signals. 
+This keeps effects primed to detect changes in the signals they are tied to.
+- **Activation**: Upon recognizing a change in an observed signal, the effect will promptly execute its predefined routine. 
+This can involve tasks like updating the UI, triggering a process, or initiating external functions. 
+Through this activation, effects ensure that the system's reactivity is both immediate and relevant.
+
+## Building a reactive system
+
+Reactive systems are designed around the principle of data-driven reactivity. This is when signals and their dependents, effects, maintain a relationship. This relationship ensures a seamless propagation of data changes throughout the system, guaranteeing its responsiveness and real-time updates.
+
+### 1. Initialization
+
+The start of a reactive system begins with the setup of signals. During this phase, signals are initialized with their respective starting values. These initialized signals serve as the primary data sources that the rest of the system will rely on.
+
+### 2. Subscription
+
+Once signals are established, the next step involves the effects indicating their dependencies. 
+In this phase, effects "subscribe" to specific signals, marking their interest and establishing a link. 
+This subscription keeps the effects aware of any changes to the signals they are tied to.
+
+### 3. Notification
+
+During the notification phase, any alteration in a signal's value sets off a chain reaction. 
+The altered signal proactively notifies all its subscribed effects about the modification. 
+Upon receiving this notification, these effects then update or activate as necessary to keep the system's state current and synchronized with the latest data changes.
+
+## Synchronous vs. asynchronous
+
+Reactivity in systems can manifest primarily in two ways: synchronous and asynchronous. 
+The choice between these often depend on the requirements of the system and the nature of the tasks involved.
+
+### Synchronous reactivity
+Synchronous reactivity operates in a direct and linear fashion. 
+When a signal changes, its corresponding effects are immediately updated in a sequential manner. 
+This form of reactivity provides a direct correlation between a signal's change and the system's response, guaranteeing that the system remains consistently aligned with the most recent data.
+
+### Asynchronous reactivity
+Asynchronous reactivity introduces a layer of flexibility and adaptability. 
```

Comment:
A "layer of flexibility and adaptability" sounds strange to me, like I'm not sure what it's saying. Asynchronous/laziness is 100% about performance basically, so it's probably simpler to just say that "it can be faster in some cases" or something.